infer/crates/kanalizer-rs/src/layers.rs

```diff
@@ -258,6 +258,33 @@ impl Gru {
     }
 }
 
+#[derive(Debug)]
+pub(crate) struct LayerNorm {
+    weight: ndarray::Array1<f32>,
+    bias: ndarray::Array1<f32>,
+    eps: f32,
+}
+
+impl LayerNorm {
+    pub(crate) fn new(weight: ndarray::Array1<f32>, bias: ndarray::Array1<f32>, eps: f32) -> Self {
+        Self { weight, bias, eps }
+    }
+
+    pub(crate) fn forward(&self, input: &ndarray::ArrayView2<f32>) -> ndarray::Array2<f32> {
+        let mean = input.mean_axis(ndarray::Axis(1)).unwrap();
+        let var = input.var_axis(ndarray::Axis(1), 0.0);
+        let std = (var + self.eps).mapv(f32::sqrt);
+
+        let mean = mean.insert_axis(ndarray::Axis(1));
+        let std = std.insert_axis(ndarray::Axis(1));
+        let norm = (input - &mean) / &std;
+        let weight = self.weight.clone().insert_axis(ndarray::Axis(0));
+        let bias = self.bias.clone().insert_axis(ndarray::Axis(0));
+
+        norm * &weight + &bias
+    }
```

Comment:
だいぶ自信ないけど、AI君に聞くにここが`Axis(1)`になっていて、実際のLayerNormは最後のaxisを正規化するから違うのでは的なこと言ってました！！
```suggestion
    pub(crate) fn forward(&self, input: &ndarray::ArrayView2<f32>) -> ndarray::Array2<f32> {
        let feature_axis = ndarray::Axis(input.ndim() - 1);

        let mean = input.mean_axis(feature_axis).unwrap();
        let var = input.var_axis(feature_axis, 0.0);
        let std = (var + self.eps).mapv(f32::sqrt);

        let mean = mean.insert_axis(feature_axis);
        let std = std.insert_axis(feature_axis);
        let norm = (input - &mean) / &std;

        let weight = self.weight.clone().insert_axis(ndarray::Axis(0));
        let bias = self.bias.clone().insert_axis(ndarray::Axis(0));

        norm * &weight + &bias
    }
```

ま、まあ、一旦可能性として。。。